pandas/tests/util/test_assert_index_equal.py

```diff
@@ -32,9 +32,9 @@ def test_index_equal_levels_mismatch():
 
 
 def test_index_equal_values_mismatch(check_exact):
-    msg = """MultiIndex level \\[1\\] are different
+    msg = """Index level \\[1\\] are different
 
-MultiIndex level \\[1\\] values are different \\(25\\.0 %\\)
+Index level \\[1\\] values are different \\(25\\.0 %\\)
```

Comment:
I'm not sure I really like this change. I think `MultiIndex level` is clearer here

Reply:
I see. Just to be on the same page, have you got the opportunity to skim #59440?

Reply:
Ok so I tried to make a compromise (https://github.com/pandas-dev/pandas/pull/59460/commits/09a1ae5fdb0585c99c51bdb897d79434f1d63125), which in short is:

```python
root_obj = "MultiIndex" if obj == "Index" else obj
...
lobj = f"{root_obj} level [{level}]"
```

I find that a little brittle, as it is based on the fact that about a hundred lines before there is `obj: str = "Index"`.

...So I tried harder (https://github.com/pandas-dev/pandas/pull/59460/commits/ad99297c4edac9a7d96bab347f422e891f2a47b0), which correspond to the current state of the PR. That is having the `obj` in the `assert_index_equal` function definition being `None` by default and branching immediately:

```python
    if obj is None:
        obj = "MultiIndex" if isinstance(left, MultiIndex) else "Index"
```

Thoughts? Better ideas?

Reply:
Hello @mroeschke, have you got the chance to check this? Thanks!

Reply:
> I'm not sure I really like this change. I think `MultiIndex level` is clearer here

No disagreement that it is slightly clearer, but it bucks the otherwise consistent behavior of `assert_.*_equal` utilizing `obj` when passed.